core/src/text/fontContext.cpp

```diff
@@ -173,21 +173,40 @@ void FontContext::renderUpdate(void* _userPtr, int* _rect, const unsigned char*
     fontContext->m_atlas->setSubData(subdata, xoff, yoff, width, height);
 }
 
-void FontContext::fontstashError(void* uptr, int error, int val) {
-    switch(error) {
-    case FONS_ATLAS_FULL:
-        LOGE("Texture Atlas full!");
-        break;
+void FontContext::fontstashError(void* _uptr, int _error, int _val) {
+    FontContext* fontContext = static_cast<FontContext*>(_uptr);
+
+    switch(_error) {
+    case FONS_ATLAS_FULL: {
+        const auto& tex = fontContext->m_atlas;
+        unsigned int nw = tex->getWidth() * 2;
+        unsigned int nh = tex->getHeight() * 2;
 
+        if (nw > TANGRAM_MAX_TEXTURE_WIDTH || nh > TANGRAM_MAX_TEXTURE_HEIGHT) {
+            LOGE("Full font texture atlas size reached!");
+        } else {
+            if (fonsExpandAtlas(fontContext->m_fsContext, nw, nh, 1)) {
+                tex->resize(nw, nh);
+                LOGW("Texture Atlas resized to %d %d", nw, nh);
+            } else {
+                LOGE("Unexpected error while expanding the font atlas");
+            }
+        }
+        break;
+    }
     case FONS_SCRATCH_FULL:
     case FONS_STATES_OVERFLOW:
     case FONS_STATES_UNDERFLOW:
     default:
-        LOGE("Unexpected error in Fontstash %d:%d!", error, val);
+        LOGE("Unexpected error in Fontstash %d:%d!", _error, _val);
         break;
     }
 }
 
+glm::vec2 FontContext::getAtlasResolution() const {
+    return glm::vec2(m_atlas->getWidth(), m_atlas->getHeight());
+}
```

Comment:
It is unlikely, but we should return the texture size from when the atlas was bound since the resize could occur between the call to bind and getAtlasResolution